Code/Framework/AzToolsFramework/AzToolsFramework/ComponentMode/ComponentModeCollection.cpp

```diff
@@ -140,7 +140,7 @@ namespace AzToolsFramework
                 m_viewportUiHandlers.emplace_back(componentType);
             }
 
-            m_activeComponentModes.push_back(AZStd::make_pair(entityComponentIdPair, componentType));
+            m_activeComponentModes.emplace_back(ComponentModeItem{ entityComponentIdPair, componentType });
```

Comment:
update: You should be able to omit `ComponentModeItem` and just do...

```
m_activeComponentModes.emplace_back(entityComponentIdPair, componentType);
```

The arguments will then be forwarded to the constructor (to have this work you might need to add a constructor for `ComponentModeItem` but it's likely worth it